core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -271,6 +272,50 @@ private static TimeUnit validate(TimeUnit timeUnit) {
     }
   }
 
+  /** {@inheritDoc}
+   *
+   * <p>BigQuery format element reference:
+   * <a href="https://cloud.google.com/bigquery/docs/reference/standard-sql/format-elements">
+   * BigQuery Standard SQL Format Elements</a>.
+   */
+  @Override public String getFormatElement(FormatElementEnum fmtElement) {
```

Comment:
Yes I like this 🌷

Reply:
I agree with @wnob that this `switch` should go, and it should use a map. Probably the map is bi-directional. Probably the map is the parse map, and that parse map can evolve from an ImmutableMap to a larger class.

Reply:
This doesn't mean we should get rid of the `getFormatElement` method (or at least something like it) entirely right? 

A `SqlDialect` would still need to know how to get an appropriate format string from a BigQuery style one while unparsing. For example, `FORMAT_TIMESTAMP(TIMESTAMP '2012-12-02 12:05:00', '%R')` would need to unparse as`TO_CHAR( TIMESTAMP '2012-12-02 12:05:00', 'HH24:MI')` for Postgres based dialects.

As you mentioned elsewhere,  this translation should not happen in a `FormatModel` _operator_ but rather the formatting function itself. That is, we should reach into the `SqlDialect` for this element mapping while in `SqlBigQueryFormatDatetimeFunction#unparse`

For library specific functions this mapping could also be used as the parse map.

Reply:
As a followup to this PR, we can define non-standard element maps per dialect. Then, when we unparse a formatting function we would use the map to find the appropriate element string for the "standard" element. The map would be bi-directional as suggested so that it can be used as both the parse map for functions and for converting elements from one dialect to another.